src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfigurationTests.cs

```diff
@@ -147,6 +167,149 @@ public void ConfigurationOrderingTest(ConfigurationLevel level)
+            File.WriteAllText(Path.Combine(userConfigDir.FullName, "settings.json"), userSettingsContent);
+
+            // Create the initial host builder.
+            IHostBuilder builder = HostBuilderHelper.CreateHostBuilder(settings);
+
+            // Override the environment configurations to use predefined values so that the test host
+            // doesn't inadvertently provide unexpected values. Passing null replaces with an empty
+            // in-memory collection source.
+            builder.ReplaceAspnetEnvironment();
+            builder.ReplaceDotnetEnvironment();
+            builder.ReplaceMonitorEnvironment();
+
+            // Build the host and get the Urls property from configuration.
+            IHost host = builder.Build();
+            IConfiguration rootConfiguration = host.Services.GetRequiredService<IConfiguration>();
+
+            Stream stream = new MemoryStream();
+
+            using ConfigurationJsonWriter jsonWriter = new ConfigurationJsonWriter(stream);
+            jsonWriter.Write(rootConfiguration, full: !redact, skipNotPresent: false);
+            jsonWriter.Dispose();
+
+            stream.Position = 0;
+
+            using (var streamReader = new StreamReader(stream))
+            {
+                string configString = streamReader.ReadToEnd();
+
+                _outputHelper.WriteLine(configString);
+
+                CompareOutput(configString, expectedConfiguration);
+            }
+        }
+
+        private string ConstructUserSettingsJson()
+        {
+            string[] fileNames = Directory.GetFiles(Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SampleConfigurations"));
+
+            IDictionary<string, JsonElement> combinedFiles = new Dictionary<string, JsonElement>();
+
+            foreach (var fileName in fileNames)
+            {
+                IDictionary<string, JsonElement> deserializedFile = JsonSerializer.Deserialize<IDictionary<string, JsonElement>>(File.ReadAllText(fileName));
+
+                combinedFiles = combinedFiles.Union(deserializedFile).ToDictionary(pair => pair.Key, pair => pair.Value);
+            }
+
+            string generatedUserSettings = JsonSerializer.Serialize(combinedFiles);
+
+            return generatedUserSettings;
+        }
+
+        private void CompareOutput(string actual, string expected)
+        {
+            Assert.Equal(CleanWhitespace(expected), CleanWhitespace(actual));
+        }
+
+        private string ConstructExpectedOutput(bool redact)
+        {
+            Dictionary<string, string> categoryMapping = GetConfigurationFileNames(redact);
+
+            using var stream = new MemoryStream();
+            using var writer = new Utf8JsonWriter(stream);
+
+            writer.WriteStartObject();
+
+            foreach (var key in OrderedConfigurationKeys)
+            {
+                writer.WritePropertyName(key);
+
+                if (categoryMapping.ContainsKey(key))
```

Comment:
```suggestion
                if (categoryMapping.TryGetValue(key, out string fileName))
```
which then avoids the need to index into the dictionary again.